docs/proposals/volumes.md

```diff
@@ -0,0 +1,431 @@
+## Abstract
+
+A proposal for sharing volumes between containers in a pod using a pod level supplemental group.
+
+## Motivation
+
+Kubernetes volumes should be usable regardless of the UID a container runs as.  This concern cuts
+across all volume types, so the system should be able to handle them in a generalized way to provide
+uniform functionality across all volume types and lower the barrier to new plugins.
+
+Goals of this design:
+
+1.  Enumerate the different use-cases for volume usage in pods
+2.  Define the desired goal state for ownership and permission management in Kubernetes
+3.  Describe the changes necessary to acheive desired state
+
+## Constraints and Assumptions
+
+1.  When writing permissions in this proposal, `D` represents a don't-care value; example: `07D0`
+    represents permissions where the owner has `7` permissions, all has `0` permissions, and group
+    has a don't-care value
+2.  Read-write usability of a volume from a container is defined as one of:
+    1.  The volume is owned by the container's effective UID and has permissions `07D0`
+    2.  The volume is owned by the container's effective GID or one of its supplemental groups and
+        has permissions `0D70`
+3.  Volume plugins should not have to handle setting permissions on volumes
+5.  We will not support securing containers within a pod from one another
```

Comment:
I don't think this is the right way to phrase this.  Instead:

Preventing two containers within a pod from reading and writing to the same volume (by choosing container UIDs that are separate) is not a current goal.